beacon-chain/casper/validator.go

```diff
@@ -108,3 +108,29 @@ func GetShardAndCommitteesForSlot(shardCommittees []*pb.ShardAndCommitteeArray,
 	}
 	return shardCommittees[slot-lcs], nil
 }
+
+// AreAttesterBitfieldsValid validates that the length of the attester bitfield matches the attester indices
+// defined in the Crystallized State.
+func AreAttesterBitfieldsValid(attestation *pb.AttestationRecord, attesterIndices []uint32) bool {
+	// Validate attester bit field has the correct length.
+	if utils.BitLength(len(attesterIndices)) != len(attestation.AttesterBitfield) {
+		log.Debugf("attestation has incorrect bitfield length. Found %v, expected %v",
+			len(attestation.AttesterBitfield), utils.BitLength(len(attesterIndices)))
+		return false
+	}
+
+	// Valid attestation can not have non-zero trailing bits.
+	lastBit := len(attesterIndices)
+	if lastBit%8 == 0 {
```

Comment:
Don't you need to check the bit? This is just seeing if the length of indices is a multiple of 8?

Reply:
The actual bit checking happens outside of this function. This is just checking if the structure of the bitfield is valid.